clang/lib/AST/CanonBounds.cpp

```diff
@@ -109,6 +110,43 @@ static Result ComparePointers(T *P1, T *P2, bool &ordered) {
   return Result::LessThan;
 }
 
+// \brief Order two source locations based on the lexicographic
+// ordering of filenames if the two source locations are in
+// different source files, and based on line/column numbers if
+// the two source locations are in the same source file.
+Result
+Lexicographic::CompareLoc(const SourceLocation *SL1,
+                          const SourceLocation *SL2) const {
+  if (SL1 && SL2) {
```

Comment:
If `SL1` or `SL2` is null, you can early exit by moving the `llvm_unreachable` from line 146 to here.